File: Algorithms/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mapping = dict((num, i) for i, num in enumerate(nums))
        for i, num in enumerate(nums):
            gap = target - num
            if gap in mapping and mapping[gap] != i:
                return [i, mapping[gap]]

File: Algorithms/twoSum.java
import java.util.*;
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mapping = new HashMap<>();
        for (int i=0; i < nums.length; i++) {
            mapping.put(nums[i], i);
        }
        for (int i = 0; i < nums.length; i++) {
            int gap = target - nums[i];
            if (mapping.containsKey(gap) &&
                    mapping.get(gap) != i) {
                return new int[]{i, mapping.get(gap)};
            }
        }
        return null;
    }
}
